packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -0,0 +1,1063 @@
+    final int index = indexFromLocation(_localDragOffset);
+    _startedOnSelectedSegment = index == highlightedIndex;
+    pressedIndex = index;
+
+    if (_startedOnSelectedSegment) {
+      _playThumbScaleAnimation(isExpanding: false);
+    }
+  }
+
+  void _onUpdate(DragUpdateDetails details) {
+    _localDragOffset = details.localPosition;
+    final int newIndex = indexFromLocation(_localDragOffset);
+
+    if (_startedOnSelectedSegment) {
+      highlightedIndex = newIndex;
+      pressedIndex = newIndex;
+    } else {
+      pressedIndex = _hasDraggedTooFar(details) ? null : newIndex;
+    }
+  }
+
+  void _onEnd(DragEndDetails details) {
+    if (_startedOnSelectedSegment) {
+      _playThumbScaleAnimation(isExpanding: true);
+      state.didChangeSelectedViaGesture();
+    }
+
+    if (pressedIndex != null) {
+      highlightedIndex = pressedIndex;
+      state.didChangeSelectedViaGesture();
+    }
+    pressedIndex = null;
+    _localDragOffset = null;
+    _startedOnSelectedSegment = null;
+  }
+
+  void _onCancel() {
+    if (_startedOnSelectedSegment) {
+      _playThumbScaleAnimation(isExpanding: true);
+    }
+
+    _localDragOffset = null;
+    pressedIndex = null;
+    _startedOnSelectedSegment = null;
+  }
+
+  void _playThumbScaleAnimation({ @required bool isExpanding }) {
+    assert(isExpanding != null);
+
+    _thumbScaleTween = Tween<double>(begin: currentThumbScale, end: isExpanding ? 1 : _kMinThumbScale);
+    state.thumbScaleController.animateWith(_kThumbSpringAnimationSimulation);
+  }
+
+  bool _hasDraggedTooFar(DragUpdateDetails details) {
+    final Offset offCenter = details.localPosition - Offset(size.width/2, size.height/2);
+    return math.pow(math.max(0, offCenter.dx.abs() - size.width/2), 2) + math.pow(math.max(0, offCenter.dy.abs() - size.height/2), 2) > _kTouchYDistanceThreshold;
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    RenderBox child = firstChild;
+    double minWidth = 0.0;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      final double childWidth = child.getMinIntrinsicWidth(height) + 2 * _kSegmentMinPadding;
+      minWidth = math.max(minWidth, childWidth);
+      child = childParentData.nextSibling;
+    }
+    return minWidth * childCount + totalSeparatorWidth;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    RenderBox child = firstChild;
+    double maxWidth = 0.0;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      final double childWidth = child.getMaxIntrinsicWidth(height) + 2 * _kSegmentMinPadding;
+      maxWidth = math.max(maxWidth, childWidth);
+      child = childParentData.nextSibling;
+    }
+    return maxWidth * childCount + totalSeparatorWidth;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    RenderBox child = firstChild;
```

Comment:
Could this be:
```dart
return getChildrenAsList.fold<double>(0, (double minHeight, RenderBox child) {
  return math.max(minHeight, child.getMinIntrinsicHeight(width));
});
```

Here and for computeMaxIntrinsicHeight().

Reply:
`getChildrenAsList`'s documentation says
>   /// This function is useful when you need random-access to the children of
  /// this render object. If you're accessing the children in order, consider
  /// walking the child list directly.

I think the imperative way is slightly more efficient?

Reply:
I don't think the efficiency aspect will matter,  but there's no harm in leaving this code as it is. It's pretty easy to read.